pkg/client/cloudevents/grpcsource/util.go

```diff
@@ -86,6 +90,88 @@ func ToManifestWork(rb *openapi.ResourceBundle) (*workv1.ManifestWork, error) {
 	return work, nil
 }
 
+func ToLabelSearch(opts metav1.ListOptions) (labels.Selector, string, bool, error) {
+	if len(opts.LabelSelector) == 0 {
+		return labels.Everything(), "", false, nil
+	}
+
+	labelSelector, err := labels.Parse(opts.LabelSelector)
+	if err != nil {
+		return nil, "", false, fmt.Errorf("invalid labels selector %q: %v", opts.LabelSelector, err)
+	}
+
+	requirements, selectable := labelSelector.Requirements()
+	if !selectable {
+		return labels.Everything(), "", false, nil
+	}
+
+	equalsLabels := []string{}
+	notEqualsLabels := []string{}
+
+	existsKeys := []string{}
+	doesNotExistKeys := []string{}
+
+	inLabels := map[string][]string{}
+
+	// refer to below links to find how to use the label selector in kubernetes
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#equality-based-requirement
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement
+	for _, requirement := range requirements {
+		switch requirement.Operator() {
+		case selection.Equals, selection.DoubleEquals:
+			values := requirement.Values()
+			if len(values) != 1 {
+				return nil, "", false, fmt.Errorf("too many values in equals operation")
+			}
+
+			equalsLabels = append(equalsLabels, fmt.Sprintf(`"%s":"%s"`, requirement.Key(), values.List()[0]))
+		case selection.NotEquals:
+			values := requirement.Values()
+			if len(values) != 1 {
+				return nil, "", false, fmt.Errorf("too many values in not equals operation")
+			}
+
+			notEqualsLabels = append(notEqualsLabels, fmt.Sprintf(`payload->'metadata'->'labels'->>'%s'<>'%s'`,
+				requirement.Key(), values.List()[0]))
```

Comment:
maybe extract `payload->'metadata'->'labels'` as a constant since it is using in many places.

Reply:
done